metabase/metabase
# Fix click actions perf issue with charts

Fixes https://github.com/metabase/metabase/issues/35206

The performance issue exists because we call `availableDrillThrus` via `visualizationIsClickable` in most visualizations during rendering, and this function can take some time to execute. This PR fixes most `availableDrillThrus` calls by moving them to event handlers. The only place that is not fixed is rendering table header cells, but there the fix is not trivial and the results are cached, unlike the places fixed in this PR.

How to verify:
- Create a dashboard with charts of all types affected in this PR
- Check that `availableDrillThrus` is not called during rendering of the dashboard

<img width="1727" alt="Screenshot 2023-10-30 at 15 21 53" src="https://github.com/metabase/metabase/assets/8542534/a0404977-81be-47f9-a489-26b54d98845b">
